manifests/init.pp

```diff
@@ -1,60 +1,424 @@
+#   Used in: templates/service/pam.erb
+#
+# @param pam_pwd_expiration_warning [Integer]
+#   Sets pam_pwd_expiration_warning in the pam section of the config file.
+#   Default value: 7
+#   Used in: templates/service/pam.erb
+#
+# @param pam_trusted_users [Optional[String]]
+#   Sets pam_trusted_users in the pam section of the config file.
+#   Default value: null
+#   Used in: templates/service/pam.erb
+#
+# @param pam_public_domains [Optional[String]]
+#   Sets pam_public_domains in the pam section of the config file.
+#   Default value: null
+#   Used in: templates/service/pam.erb
+#
+# @param reconnection_retries [Integer]
+#   Sets the value of reconnection_retries in the nss, pam, and sssd sections
+#   of the sssd config file.
+#   Default value: 3
+#   Used in: templates/service/pam.erb
+#            templates/service/nss.erb
+#            templates/sssd.conf.erb
+#
+# @param re_expression [Optional[String]]
+#   Sets re_expression in the domain and sssd sections of the config file.
+#   Default value: null
+#   Used in: templates/sssd.conf.erb
+#            templates/domain.erb
+#
+# @param services [Sssd::Services]
+#   Sets the value of services in the sssd section of the config file.
+#   Default value: ['nss','pam','ssh','sudo']
+#   Used in: templates/sssd.conf.erb
+#
+# @param ssh_hash_known_hosts [Boolean]
+#   Sets the value of ssh_hash_known_hosts in the ssh section of the config
+#   Default value: true
+#   Used in: templates/service/ssh.erb
+#
+# @param ssh_known_hosts_timeout [Optional[Integer]]
+#   Sets the value of sssh_known_hosts_timeout in the ssh section of config
+#   Default value: null
+#   Used in: templates/service/ssh.erb
+#
+# @param sudo_timed [Boolean]
+#   Sets the value of sudo_timed in the sudo section of the config file.
+#   Default value: false
+#   Used in: templates/service/sudo.erb
+#
+# @param try_inotify [Boolean]
+#   Sets the try_inotify field in the sssd section of the conf file
+#   Default value: true
+#   Used in: templates/service/ssh.conf.erb
+#
+# @param user [Optional[String]]
+#   Sets the value of the user field in the sssd section of the conf file.
+#   Default value: null
+#   Used in: templates/sssd.conf.erb
+#
+# @param user_attributes [Optional[String]]
+#   Sets the user_attributes field in the nss section of the conf file
+#   Default value: null
+#   Used in: templates/service/nss.erb
+#
+# @param vetoed_shells [Optional[String]]
+#   Sets the value of the vetoed_shells field in the nss section of sssd conf
+#   Default value: null
+#   Used in: templates/service/nss.erb
+#
+
 class sssd (
+
+  ## These variables control the initial installation and configuration
+  ## of the sssd package and its dependencies.
+  Optional[String]               $client_package_name,
+  Optional[String]               $client_package_version,
+  Stdlib::Absolutepath           $conf_file_path,
+  Optional[String]               $conf_dir_owner,
+  Stdlib::Absolutepath           $conf_dir_path,
```

Comment:
Do you mean that not having defaults set in the manifest is going to prevent the console from auto-filling default values? I'm not sure I understand what you mean when you say "this is going to render the console useless for these values". What is 'this' in this context?